Content.Server/Storage/ItemCounterSystem.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Content.Server.Storage.Components;
+using Content.Shared.Storage;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+
+namespace Content.Server.Storage
+{
+    [UsedImplicitly]
+    public class ItemCounterSystem : SharedItemCounterSystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SharedItemCounterComponent, ComponentInit>(OnComponentInit);
+        }
+
+        private void OnComponentInit(EntityUid uid, SharedItemCounterComponent component, ComponentInit args)
+        {
+            if (!ComponentManager.TryGetComponent(uid, out SharedAppearanceComponent? appearance)
+                || !component.Owner.TryGetComponent(out ServerStorageComponent? _))
```

Comment:
just use componentmanager for both of these checks, idk why you only did it for one

also do HasComponent for the second one rather than TryGet